File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <stack>

using namespace std;

stack<int> st;
bool solution(string s)
{
    bool answer = true;
    
    for(char c : s){
        if(c == '('){
            st.push(1);
        }
        else if(c == ')'&& !st.empty()){
            st.pop();
        }
        else return false;
        
    }
    
    if(!st.empty()) return false;

    return true;;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        Stack<Integer> stack = new Stack<>();
        
        for(int i=0; i<s.length(); i++){
            if(s.charAt(i) == '(') stack.push(1);
            else if(s.charAt(i) == ')'){
                if(stack.isEmpty()) return false;
                stack.pop();
            }
        }
        if(!stack.isEmpty()) return false;
        return true;
    }
}